autogpt_platform/backend/backend/blocks/wavespeed/ai_media_generator_test.py

```
@@ -0,0 +1,277 @@
+        await AIMediaGeneratorBlock().generate_media(_input(), TEST_CREDENTIALS)
+
+
+@pytest.mark.asyncio
+async def test_generate_media_gives_up_after_the_polling_budget():
+    post = AsyncMock(return_value=_response(_submitted()))
+    get = AsyncMock(return_value=_response({"data": {"status": "processing"}}))
+    patcher, _ = _patch_requests(post, get)
+    with patcher, patch(
+        "backend.blocks.wavespeed.ai_media_generator.asyncio.sleep", new=AsyncMock()
+    ):
+        with pytest.raises(RuntimeError, match="Maximum polling attempts reached"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)